extensions/applicationinsights-properties-js/src/Context/User.ts

```diff
@@ -61,10 +61,14 @@ export class User implements IUserContext {
     constructor(config: ITelemetryConfig, core: IAppInsightsCore) {
         let _logger = safeGetLogger(core);
         let _cookieManager: ICookieMgr = safeGetCookieMgr(core);
+        let _storageNamePrefix: () => string;
 
         dynamicProto(User, this, (_self) => {
+            _self.config = config;
```

Comment:
Have following questions:
1. In the docs , its mentioned that namePrefix is expected to be a string, but here it looks like callback function.
2. Also does the namePrefix have to be appended to all cookies? or is it only ai_user and ai_session?
3. nit: its confusing to use namePrefix in configuration, but its actually used as postfix

Reply:
1. We wrap it internall with a function, so the value passed in from the config is a string
2. I have the same question as we have existing users already deployed with this and this cookie is LOOOOONG lasting, we should not automatically add it -- there should be a config otherwise we are going to cause an issue with anyone that is using this. Also, by not including the prefix on this cookie it can (and is) being shared across multiple sites.
3. Agree, but its not easy to change this we could introduce an new config value which defaults to re-using the namePrefix -- but this assumes we can stop using namePrefix which we cant as it's also used for storage.
- Options, add new configs ```cookiePostfix``` (defaults to the same as namePrefix if not provided, so the ai_session keeps working) and either ```prefixUserCookie:boolean``` which defaults to false or split them ```userCookiePostfix``` which defaults to an empty string, for this later one would recommend using ```sessionCookiePostfix``` instead of a generic ```cookiePostfix```